File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/RadioButton_ViewTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "RadioButton_ViewModelTestEnvironment.h"
#include <string>
#include "convertOptionsToString.h"
#include "RadioButton_ViewModel.h"
#include "RadioButton_ViewModelOptionsOption.h"
#include "RadioButton_ViewModelTestEnvironmentImpl.h"

namespace widgetassertions
{
  class RadioButton_ViewTests : public testing::Test
  {
  public:
    std::shared_ptr<RadioButton_ViewModel> sut;
    std::shared_ptr<RadioButton_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    virtual void then_Options_selected_OptionB_();
    virtual void then_Options_selected_OptionA_and_OptionB_is_not_enabled();
    virtual void then_Options_selected_OptionA_and_OptionB_is_not_visible();
  protected:
    void SetUp() override;
  };
  void RadioButton_ViewTests::SetUp()
  {
    this->testEnvironment = std::make_shared<RadioButton_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  TEST_F(RadioButton_ViewTests, Select_Option_given_when_then_Options_selected_OptionB_)
  {
    this->BuildSut();
    this->then_Options_selected_OptionB_();
  }
  TEST_F(RadioButton_ViewTests, RadioButton_disabled_given_when_then_Options_selected_OptionA_and_OptionB_is_not_enabled)
  {
    this->BuildSut();
    this->then_Options_selected_OptionA_and_OptionB_is_not_enabled();
  }
  TEST_F(RadioButton_ViewTests, RadioButton_invisible_given_when_then_Options_selected_OptionA_and_OptionB_is_not_visible)
  {
    this->BuildSut();
    this->then_Options_selected_OptionA_and_OptionB_is_not_visible();
  }
  void RadioButton_ViewTests::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void RadioButton_ViewTests::then_Options_selected_OptionB_()
  {
    EXPECT_EQ(widgetassertions::RadioButton_ViewModelOptionsOption::OptionB, this->sut->getOptionsRadioButtonGroupSelectedEntry()) << std::string("Expected that radio button group Options has selected <") + std::string("OptionB") + std::string(">, but was <") + widgetassertions::convertOptionsToString(this->sut->getOptionsRadioButtonGroupSelectedEntry()) + std::string(">");
  }
  void RadioButton_ViewTests::then_Options_selected_OptionA_and_OptionB_is_not_enabled()
  {
    EXPECT_EQ(widgetassertions::RadioButton_ViewModelOptionsOption::OptionA, this->sut->getOptionsRadioButtonGroupSelectedEntry()) << std::string("Expected that radio button group Options has selected <") + std::string("OptionA") + std::string(">, but was <") + widgetassertions::convertOptionsToString(this->sut->getOptionsRadioButtonGroupSelectedEntry()) + std::string(">");
    EXPECT_FALSE(this->sut->getIsOptionBRadioButtonEnabled()) << std::string("Expected radio button OptionB is disabled, but it was enabled");
  }
  void RadioButton_ViewTests::then_Options_selected_OptionA_and_OptionB_is_not_visible()
  {
    EXPECT_EQ(widgetassertions::RadioButton_ViewModelOptionsOption::OptionA, this->sut->getOptionsRadioButtonGroupSelectedEntry()) << std::string("Expected that radio button group Options has selected <") + std::string("OptionA") + std::string(">, but was <") + widgetassertions::convertOptionsToString(this->sut->getOptionsRadioButtonGroupSelectedEntry()) + std::string(">");
    EXPECT_FALSE(this->sut->getIsOptionBRadioButtonVisible()) << std::string("Expected radio button OptionB is hidden, but it was visible");
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/RadioButton_ViewTests.cs
/// <filename>
///     RadioButton_ViewTests.cs
/// </filename>
using widgetassertions;
using Microsoft.VisualStudio.TestTools.UnitTesting;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class RadioButton_ViewTests
{
    private widgetassertions.RadioButton_ViewModel sut;

    private widgetassertions.RadioButton_ViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new widgetassertions.RadioButton_ViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void Select_Option_given_when_then_Options_selected_OptionB_()
    {
        this.BuildSut();
        this.then_Options_selected_OptionB_();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void RadioButton_disabled_given_when_then_Options_selected_OptionA_and_OptionB_is_not_enabled()
    {
        this.BuildSut();
        this.then_Options_selected_OptionA_and_OptionB_is_not_enabled();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void RadioButton_invisible_given_when_then_Options_selected_OptionA_and_OptionB_is_not_visible()
    {
        this.BuildSut();
        this.then_Options_selected_OptionA_and_OptionB_is_not_visible();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    public virtual void then_Options_selected_OptionB_()
    {
        Assert.AreEqual(widgetassertions.RadioButton_ViewModelOptionsOption.OptionB, this.sut.getOptionsRadioButtonGroupSelectedEntry(), "Expected that radio button group Options has selected <" + "OptionB" + ">, but was <" + ActivityconvertOptionsToString.convertOptionsToString(this.sut.getOptionsRadioButtonGroupSelectedEntry()) + ">");
    }

    public virtual void then_Options_selected_OptionA_and_OptionB_is_not_enabled()
    {
        Assert.AreEqual(widgetassertions.RadioButton_ViewModelOptionsOption.OptionA, this.sut.getOptionsRadioButtonGroupSelectedEntry(), "Expected that radio button group Options has selected <" + "OptionA" + ">, but was <" + ActivityconvertOptionsToString.convertOptionsToString(this.sut.getOptionsRadioButtonGroupSelectedEntry()) + ">");
        Assert.IsFalse(this.sut.getIsOptionBRadioButtonEnabled(), "Expected radio button OptionB is disabled, but it was enabled");
    }

    public virtual void then_Options_selected_OptionA_and_OptionB_is_not_visible()
    {
        Assert.AreEqual(widgetassertions.RadioButton_ViewModelOptionsOption.OptionA, this.sut.getOptionsRadioButtonGroupSelectedEntry(), "Expected that radio button group Options has selected <" + "OptionA" + ">, but was <" + ActivityconvertOptionsToString.convertOptionsToString(this.sut.getOptionsRadioButtonGroupSelectedEntry()) + ">");
        Assert.IsFalse(this.sut.getIsOptionBRadioButtonVisible(), "Expected radio button OptionB is hidden, but it was visible");
    }
}
